File: enterprise/stock_barcode/models/stock_location.py
# -*- coding: utf-8 -*-
# Part of Odoo. See LICENSE file for full copyright and licensing details.

from odoo import models, api


class Location(models.Model):
    _inherit = 'stock.location'
    _barcode_field = 'barcode'

    @api.model
    def _search(self, domain, offset=0, limit=None, order=None):
        domain = self.env.company.nomenclature_id._preprocess_gs1_search_args(domain, ['location', 'location_dest'])
        return super()._search(domain, offset=offset, limit=limit, order=order)

    @api.model
    def _get_fields_stock_barcode(self):
        return ['barcode', 'display_name', 'name', 'parent_path', 'usage']

    def get_counted_quant_data_records(self):
        self.ensure_one()
        return self.quant_ids.get_stock_barcode_data_records()

File: enterprise/stock_barcode_product_expiry/static/tests/mock_models/stock_location.js
import { fields, models } from "@web/../tests/web_test_helpers";

export class StockLocation extends models.Model {
    name = fields.Char();
    barcode = fields.Char();
    parent_path = fields.Char();
    _records = [
        {
            id: 5,
            barcode: "WH-STOCK 1",
            display_name: "WH/STOCK 1",
            name: "STOCK 1",
            parent_path: "1/7",
        },
        {
            id: 6,
            barcode: "WH-STOCK 2",
            display_name: "WH/STOCK 2",
            name: "STOCK 2",
            parent_path: "1/7/8",
        },
    ];
}
